examples/generation-based-subscription/src/components/navbar.tsx

```diff
@@ -0,0 +1,187 @@
+'use client';
+
+import { useState } from 'react';
+import { useRouter } from 'next/navigation';
+import { useQueryClient } from '@tanstack/react-query';
+import { authClient } from '@/lib/auth-client';
+import { useBilling } from '@flowglad/nextjs';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  DropdownMenuTrigger,
+} from '@/components/ui/dropdown-menu';
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from '@/components/ui/tooltip';
+import { Button } from '@/components/ui/button';
+
+export function Navbar() {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+  const { data: session } = authClient.useSession();
+  const billing = useBilling();
+  const [isCancelling, setIsCancelling] = useState(false);
+  const [cancelError, setCancelError] = useState<string | null>(null);
+
+  async function handleSignOut() {
+    await queryClient.clear();
+    await authClient.signOut({
+      fetchOptions: {
+        onSuccess: () => {
+          router.push('/sign-in');
+        },
+      },
+    });
+  }
+
+  async function handleCancelSubscription() {
+    // By default, each customer can only have one active subscription at a time,
+    // so accessing the first currentSubscriptions is sufficient.
+    // Multiple subscriptions per customer can be enabled in dashboard > settings
+    const currentSubscription = billing.currentSubscriptions?.[0];
+    const subscriptionId = currentSubscription?.id;
+
+    if (!subscriptionId || !billing.cancelSubscription) {
+      return;
+    }
+
+    // Confirm cancellation
+    const confirmed = window.confirm(
+      'Are you sure you want to cancel your membership? Your subscription will remain active until the end of the current billing period.'
+    );
+
+    if (!confirmed) {
+      return;
+    }
+
+    setIsCancelling(true);
+    setCancelError(null);
+
+    try {
+      await billing.cancelSubscription({
+        id: subscriptionId,
+        cancellation: {
+          timing: 'at_end_of_current_billing_period',
+        },
+      });
+    } catch (error) {
+      setCancelError(
+        error instanceof Error
+          ? error.message
+          : 'Failed to cancel subscription. Please try again.'
+      );
+    } finally {
+      setIsCancelling(false);
+    }
+  }
+
+  if (!session?.user) {
+    return null;
+  }
+
+  const accountName = session.user.name || session.user.email || 'Account';
+  // By default, each customer can only have one active subscription at a time,
+  // so accessing the first currentSubscriptions is sufficient.
+  // Multiple subscriptions per customer can be enabled in dashboard > settings
+  const currentSubscription = billing.currentSubscriptions?.[0];
+
+  // Check if subscription is a default plan (cannot be cancelled)
+  // Default plans have default: true at the product level OR isDefault: true at the price level
```

Comment:
remove "OR isDefault: true at the price level", as the latest price is always set as the default now for price immutability, and it will be consistent with the comment below "Only check product.default, not price.isDefault"
```suggestion
  // Default plans have default: true at the product level
```